config-store-yaml/src/test/java/io/stargate/config/store/yaml/ConfigStoreActivatorYamlDoesNotExistsTest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.config.store.yaml;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import io.stargate.config.store.ConfigStore;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.osgi.framework.BundleContext;
+
+class ConfigStoreActivatorYamlDoesNotExistsTest {
+
+  @BeforeAll
+  public static void setup() {
+    System.setProperty("stargate.config_store.yaml.location", "non_existing_path");
+  }
+
+  @Test
+  public void shouldNotRegisterConfigStoreWhenYamlLocationHasNotExistingStargateConfig() {
+    // given
+    BundleContext bundleContext = mock(BundleContext.class);
+    ConfigStoreActivator kafkaProducerActivator = new ConfigStoreActivator();
+
+    // when
+    kafkaProducerActivator.start(bundleContext);
+
+    // then
+    verify(bundleContext, times(0))
+        .registerService(eq(ConfigStore.class), any(ConfigStoreYaml.class), any());
```

Comment:
Should we have a test for starting with no config file? I think it's worth adding it since we apparently expect this situation to happen at some point.

Reply:
The test in this class validates exactly this case.  The file does not exist - there is no config file.

Reply:
you're right, I got confused, sorry about the noise :facepalm: